### PR TITLE
GRAPHICS: Remove use of TransparentSurface in MacGUI code

### PR DESCRIPTION
**Pull request by ccawley2011**
The NinePatch code only needed colour key transparency instead of full alpha blending, so it should be faster to use `ManagedSurface::blitFrom` with a paletted source instead of using `blendBlitFrom` and extra intermediate surfaces.

Only the WAGE engine has been tested with these changes.

### CODE REVIEW

File: graphics/nine_patch.cpp [hunk 1]
```diff
@@ -186,13 +184,43 @@ NinePatchBitmap::NinePatchBitmap(Graphics::TransparentSurface *bmp, bool owns_bi
...
 
+	const uint32 key = bmp->getTransparentColor();
+	uint32 black, white;
+
+	if (bmp->format.isCLUT8()) {
+		uint8 palette[256 * 3];
+		bmp->grabPalette(palette, 0, 256);
+
+		black = 0;
+		white = 1;
```
Comment (OMGPizzaGuy) [inline graphics/nine_patch.cpp:195]: Should these two be set to uint32(-1) for following bad_bitmap checks?

Comment (ccawley2011) [reply graphics/nine_patch.cpp]: Good catch. Fixed.

**Comment by sev-** [comment]
I checked Director, MacVenture, Pink WAGE engines. No regressions. Merging this.

**Comment by sev-** [comment]
Great work!

**Pull request merged**